crates/valence/examples/chat.rs

```diff
@@ -0,0 +1,115 @@
+use bevy_app::App;
+use tracing::warn;
+use valence::client::despawn_disconnected_clients;
+use valence::client::event::{default_event_handler, ChatCommand, ChatMessage};
+use valence::prelude::*;
+
+const SPAWN_Y: i32 = 64;
+const ARENA_RADIUS: i32 = 32;
+
+pub fn main() {
+    tracing_subscriber::fmt().init();
+
+    App::new()
+        .add_plugin(ServerPlugin::new(()))
+        .add_startup_system(setup)
+        .add_system_to_stage(EventLoop, default_event_handler)
+        .add_system_to_stage(EventLoop, handle_message_events)
+        .add_system_to_stage(EventLoop, handle_command_events)
+        .add_system(init_clients)
+        .add_system(despawn_disconnected_clients)
+        .add_system_set(PlayerList::default_system_set())
+        .add_system(teleport_oob_clients)
+        .run();
+}
+
+fn setup(world: &mut World) {
+    let mut instance = world
+        .resource::<Server>()
+        .new_instance(DimensionId::default());
+
+    for z in -5..5 {
+        for x in -5..5 {
+            instance.insert_chunk([x, z], Chunk::default());
+        }
+    }
+
+    // Create circular arena.
+    for z in -ARENA_RADIUS..ARENA_RADIUS {
+        for x in -ARENA_RADIUS..ARENA_RADIUS {
+            let dist = f64::hypot(x as _, z as _) / ARENA_RADIUS as f64;
+
+            if dist > 1.0 {
+                continue;
+            }
+
+            let block = if rand::random::<f64>() < dist {
+                BlockState::STONE
+            } else {
+                BlockState::DEEPSLATE
+            };
+
+            for y in 0..SPAWN_Y {
+                instance.set_block_state([x, y, z], block);
+            }
+        }
+    }
```

Comment:
Make this more simple. A square 1 block high would suffice.

Because this is an example, any code not related to what the example is showing off needs to be as minimal as possible.

Reply:
Actually, you can generate infinity layer of barrier block.